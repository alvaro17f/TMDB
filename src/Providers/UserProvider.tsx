import { createContext, useContext, useReducer } from "react";
import { GuestActionType, GuestType, UserContextType } from "../types/types";

const UserContext = createContext<UserContextType>({});

const useUserContext = () => {
  return useContext(UserContext);
};

const initialState = {
  success: undefined,
  guest_session_id: undefined,
  expires_at: undefined,
};

const reducer = (state: GuestType, action: GuestActionType): GuestType => {
  switch (action.type) {
    case "CREATE_GUEST_SESSION": {
      return {
        ...state,
        success: action.payload.success,
        guest_session_id: action.payload.guest_session_id,
        expires_at: action.payload.expires_at,
      };
    }
    default: {
      return state;
    }
  }
};

const UserProvider = ({ children }: { children: React.ReactNode }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <UserContext.Provider value={{ state, dispatch }}>
      {children}
    </UserContext.Provider>
  );
};

export { UserProvider, useUserContext };
