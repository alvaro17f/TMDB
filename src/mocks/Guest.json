{
  "success": true,
  "guest_session_id": "ca6c96ee6faea49df91fd2f1fd48ffed",
  "expires_at": "2023-06-01 10:32:07 UTC"
}
