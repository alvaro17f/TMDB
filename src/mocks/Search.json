{
    "page": 1,
    "results": [
      {
        "adult": false,
        "backdrop_path": "/k09W6niO9wXlKaj5G1HLQXzMxxa.jpg",
        "genre_ids": [
          35,
          18
        ],
        "id": 525938,
        "original_language": "fr",
        "original_title": "Break",
        "overview": "Lucie, a young dancer suffers an accident. When she wakes up from a coma, she and her partner start training for a dance show. On the way, she starts looking for her estranged father, a path that might lead her to life-changing love.",
        "popularity": 8.733,
        "poster_path": "/qwK4PMUppMKlm2CwspO8EKuhj7y.jpg",
        "release_date": "2018-07-18",
        "title": "Break",
        "video": false,
        "vote_average": 6.003,
        "vote_count": 156
      },
      {
        "adult": false,
        "backdrop_path": "/5apdF3UVO7x65VYkQ1iSDlWwrkG.jpg",
        "genre_ids": [
          80,
          18
        ],
        "id": 661738,
        "original_language": "en",
        "original_title": "Break",
        "overview": "Break is a 'rags to riches', feel good story. The film follows the exploits of Spencer Pryde (Sam Gittins), a gifted, inner-city kid, wasting his talents on petty crime. After witnessing the brutal murder of his pal Denis and finding himself in debt to a drug dealing thug named Ginger, it seems Spencer's life is spiraling out of control, until one day, a chance encounter with a Chinese stranger and former eight-ball pool champion named Vincent Quiang (David Yip) presents him with an opportunity to turn his life around. But in order to make a new life for himself, Spencer will first need to break away from peer pressure of his friends, his environment and all the negative influences of his current life.",
        "popularity": 3.895,
        "poster_path": "/m3wc1fnDyJjw0hiuMN9jxw9s7tI.jpg",
        "release_date": "2020-04-20",
        "title": "Break",
        "video": false,
        "vote_average": 5.3,
        "vote_count": 10
      },
      {
        "adult": false,
        "backdrop_path": "/sNuRc3ALFS4KjgjmiypZJoEYBG5.jpg",
        "genre_ids": [
          28,
          27
        ],
        "id": 55939,
        "original_language": "en",
        "original_title": "Break",
        "overview": "Four young girls head out into the Canadian woods for some outdoors fun. They wind up running into 2 brutal and sadistically inhuman rednecks. A blood-soaked fight for survival ensues.",
        "popularity": 2.958,
        "poster_path": "/vBss1480NrIhLzGaFtIqk8xcZuc.jpg",
        "release_date": "2009-10-02",
        "title": "Break",
        "video": false,
        "vote_average": 3.4,
        "vote_count": 16
      },
      {
        "adult": false,
        "backdrop_path": "/gnHCbbkmK8siSHFUax606g77eih.jpg",
        "genre_ids": [
          28,
          80,
          18,
          53
        ],
        "id": 20750,
        "original_language": "en",
        "original_title": "Break",
        "overview": "A terminally ill crime boss, known only as \"The Man\", hires a hit man named Frank to carry out his own assassination as well as the assassination of \"The Woman\" he loves. When \"The Woman\" turns out to be Frank's long lost lover, he turns against \"The Man\" becoming the target himself.",
        "popularity": 1.88,
        "poster_path": "/xETSL8BFFMGxsifiuKJ52AOJeMR.jpg",
        "release_date": "2009-05-01",
        "title": "Break",
        "video": false,
        "vote_average": 4.5,
        "vote_count": 11
      },
      {
        "adult": false,
        "backdrop_path": "/e9gy02r7RmZHhdwS7sAxc0huhDV.jpg",
        "genre_ids": [
          18
        ],
        "id": 958718,
        "original_language": "en",
        "original_title": "Break",
        "overview": "A young Japanese-American man crosses the line with the underworld element in Tokyo, while dealing with his own personal struggle of being an outsider trying to fit in with Japanese society.",
        "popularity": 0.84,
        "poster_path": "/nuU1aeN7GneAMYpbVNef6GPqJJm.jpg",
        "release_date": "2022-03-01",
        "title": "Break",
        "video": false,
        "vote_average": 0,
        "vote_count": 0
      },
      {
        "adult": false,
        "backdrop_path": null,
        "genre_ids": [
          35
        ],
        "id": 703108,
        "original_language": "en",
        "original_title": "BREAK.",
        "overview": "AG, a pragmatic introvert, is forced to leave the big city and return to his old hometown in the rural midwest. Sparks fly when he reunites with Marli, his idealistic childhood pal. They spend their vacation at the local roller rink, getting buzzed and into trouble as the dog days of summer melt away. But what happens when summer ends?",
        "popularity": 1.738,
        "poster_path": "/gjIJd4FYsw1DXZF9EBuKVosqi4i.jpg",
        "release_date": "2020-05-11",
        "title": "BREAK.",
        "video": false,
        "vote_average": 10,
        "vote_count": 1
      },
      {
        "adult": false,
        "backdrop_path": null,
        "genre_ids": [
          18
        ],
        "id": 653591,
        "original_language": "en",
        "original_title": "Break",
        "overview": "Following an initial misunderstanding between an older couple and a young family, fate takes a hand in bringing both parties closer together. One small act of kindness is then reciprocated by a very big one.",
        "popularity": 0.875,
        "poster_path": "/aqG4opjPrVVPh8rFm8nHTroGOWy.jpg",
        "release_date": "2015-05-01",
        "title": "Break",
        "video": false,
        "vote_average": 7,
        "vote_count": 1
      },
      {
        "adult": false,
        "backdrop_path": null,
        "genre_ids": [
          18
        ],
        "id": 944691,
        "original_language": "en",
        "original_title": "Break",
        "overview": "A young Japanese-American man crosses the line with the underworld element in Tokyo, while dealing with his own personal struggle of being an outsider trying to fit in with Japanese society.",
        "popularity": 0.679,
        "poster_path": null,
        "release_date": "2022-02-28",
        "title": "Break",
        "video": false,
        "vote_average": 0,
        "vote_count": 0
      },
      {
        "adult": false,
        "backdrop_path": null,
        "genre_ids": [],
        "id": 636144,
        "original_language": "en",
        "original_title": "Break",
        "overview": "The hardworking people of this seemingly quiet town return home to their families after a long day of honest work. The birds do the same. On go the street lamps. It’ll be dark soon.",
        "popularity": 0.6,
        "poster_path": "/kpan5mSCm7hwmKguKBbaqqR33GN.jpg",
        "release_date": "2019-10-02",
        "title": "Break",
        "video": false,
        "vote_average": 0,
        "vote_count": 0
      },
      {
        "adult": false,
        "backdrop_path": null,
        "genre_ids": [],
        "id": 639171,
        "original_language": "en",
        "original_title": "Break",
        "overview": "A long-term relationship is ending; set in rural new Zealand a depressed mother makes the difficult decision to leave her son and partner.  The film explores the relationship a nine-year-old boy has with his mother and the ultimate loss of his trust.",
        "popularity": 0.6,
        "poster_path": null,
        "release_date": "2006-01-01",
        "title": "Break",
        "video": false,
        "vote_average": 0,
        "vote_count": 0
      },
      {
        "adult": false,
        "backdrop_path": null,
        "genre_ids": [],
        "id": 1004821,
        "original_language": "en",
        "original_title": "Break",
        "overview": "Two workers get to know each other intimately while on a smoke break outside the shipping facility they work at.",
        "popularity": 0.6,
        "poster_path": "/8VrugmuIzBxBolScrE7FIzFAQmA.jpg",
        "release_date": "2013-06-21",
        "title": "Break",
        "video": false,
        "vote_average": 0,
        "vote_count": 0
      },
      {
        "adult": false,
        "backdrop_path": null,
        "genre_ids": [],
        "id": 730410,
        "original_language": "en",
        "original_title": "Break",
        "overview": "Reunited for the weekend, a group of friends are driven to violence when an unknown force seems to bring out the worst in them. Who will be the first to break?",
        "popularity": 0.6,
        "poster_path": "/wYlsbNzPLdl96M8Oue3Wdxcc7Es.jpg",
        "release_date": "2010-01-01",
        "title": "Break",
        "video": false,
        "vote_average": 0,
        "vote_count": 0
      },
      {
        "adult": false,
        "backdrop_path": null,
        "genre_ids": [
          18,
          28,
          35,
          53
        ],
        "id": 100615,
        "original_language": "en",
        "original_title": "Break",
        "overview": "Right in the middle of a game of tennis, two childhood friends are facing.What's going on inside these two men's heads as the game comes to its end?",
        "popularity": 0.6,
        "poster_path": null,
        "release_date": "2008-05-09",
        "title": "Break",
        "video": false,
        "vote_average": 0,
        "vote_count": 0
      },
      {
        "adult": false,
        "backdrop_path": null,
        "genre_ids": [],
        "id": 222189,
        "original_language": "pt",
        "original_title": "Break",
        "overview": "André is a man with the need to pause.  In a world where life, is so futile and fragile, he just needs a break.  But what is reality, when that break... becomes too long?",
        "popularity": 0.6,
        "poster_path": "/aX3cxsAV1N8HCOerdyDpXEJSkjs.jpg",
        "release_date": "2013-09-06",
        "title": "Break",
        "video": false,
        "vote_average": 0,
        "vote_count": 0
      },
      {
        "adult": false,
        "backdrop_path": null,
        "genre_ids": [],
        "id": 779649,
        "original_language": "en",
        "original_title": "Break",
        "overview": "The short film Break is a creative collaboration between visual artist Joan Perlman and noted composer Laurie Spiegel (The Hunger Games, The Expanding Universe (CD)). Break explores the transmuting relationship of ice and water in the north Atlantic off Iceland's coast, through varying shots of a single act: glacial ice fragments violently struck by breaking waves in a stormy ocean. Images and field recordings of the ocean's relentless force paired with Spiegel's unsettling, otherworldly score evoke the troubling effects of an unseen human presence. Break is a meditation on the passage of time and fragility of matter. With powerful imagery and sound, its visual commentary alludes to inexorable loss and dissolution and infers the implications of a changing climate in the far north, and beyond.",
        "popularity": 0.6,
        "poster_path": null,
        "release_date": "",
        "title": "Break",
        "video": false,
        "vote_average": 0,
        "vote_count": 0
      },
      {
        "adult": false,
        "backdrop_path": null,
        "genre_ids": [],
        "id": 228926,
        "original_language": "en",
        "original_title": "Break",
        "overview": "A young boy seeks salvation from his girlfriend when he is turned down from the one he loves it turns into something deeper and darker. We take a mind trip into their fears and dreams.",
        "popularity": 0.6,
        "poster_path": null,
        "release_date": "2004-02-29",
        "title": "Break",
        "video": false,
        "vote_average": 0,
        "vote_count": 0
      },
      {
        "adult": false,
        "backdrop_path": "/iFT1kQi0x6cuxOy7OIRCnijJnjF.jpg",
        "genre_ids": [
          35
        ],
        "id": 264967,
        "original_language": "de",
        "original_title": "Die Nichten der Frau Oberst",
        "overview": "Young widow Madame Yanne takes care of her pretty nieces Florentine and Juliet, who are currently living out their lesbian obsessions and are currently more engaged with each other than turning their heads around the lusty male world.",
        "popularity": 38.801,
        "poster_path": "/vs0uPhPfmaR17wB9uXhxiS8ZzZY.jpg",
        "release_date": "1968-07-19",
        "title": "Guess Who's Coming for Breakfast",
        "video": false,
        "vote_average": 3.7,
        "vote_count": 5
      },
      {
        "adult": false,
        "backdrop_path": "/qkl57wzSFrpi2sRpoc2mZJbMuLP.jpg",
        "genre_ids": [
          12,
          14,
          18,
          10749
        ],
        "id": 50620,
        "original_language": "en",
        "original_title": "The Twilight Saga: Breaking Dawn - Part 2",
        "overview": "After the birth of Renesmee, the Cullens gather other vampire clans in order to protect the child from a false allegation that puts the family in front of the Volturi.",
        "popularity": 55.93,
        "poster_path": "/7IGdPaKujv0BjI0Zd0m0a4CzEjJ.jpg",
        "release_date": "2012-11-13",
        "title": "The Twilight Saga: Breaking Dawn - Part 2",
        "video": false,
        "vote_average": 6.476,
        "vote_count": 8207
      },
      {
        "adult": false,
        "backdrop_path": "/2gozFsO8OfW63HcoksUEJNx1k3P.jpg",
        "genre_ids": [
          35
        ],
        "id": 387846,
        "original_language": "ko",
        "original_title": "청춘학당 2: 기생난입야사",
        "overview": "During the late period of the Joseon, Jae-won, Man-ki, Deok-ho and foolish Chil-deuk are the only ones at the village school. The school is going through a rough patch so the school teacher visits his  girlfriend the gisaeng and suggests to his students that they bring in female students. So-hyang and the other girls brighten up the mood of the school and more male students sign up. Yeon-hwa is jealous of Jae-won and So-hyang who fell in love with each other at first sight. She confesses her love to him but he doesn't give her any attention so she tells him So-hyang's secret. On the other hand, the teacher's wife is surprised to see a man wearing a mask peeking at her body but is surprised to the see the size of the hole in the pumpkin he dropped. Not happy with her husband's night skills, she calls in the male students one by one to find out but the pumpkin belongs to no one...  To whom does the hole belong to?",
        "popularity": 7.106,
        "poster_path": "/ihUg1Xi58MWRzrj5WOgea1jvSt5.jpg",
        "release_date": "2016-03-03",
        "title": "School Of Youth 2: The Unofficial History of the Gisaeng Break-In",
        "video": false,
        "vote_average": 1.667,
        "vote_count": 3
      },
      {
        "adult": false,
        "backdrop_path": "/sEXO8VGcVrXbXYXOtKlFqsBhnba.jpg",
        "genre_ids": [
          53
        ],
        "id": 517024,
        "original_language": "ru",
        "original_title": "Отрыв",
        "overview": "The company of three guys and two girls makes their way to the old funicular to have an unforgettable New Year’s Eve celebration at the top of the mountain. A fun adventure turns into a nightmare when the funicular stops over the abyss. Coldness, height and fear of a slow death make them reveal their true essence and get in a fight with each other.",
        "popularity": 11.968,
        "poster_path": "/66OBBkqi9v5txzDPLzGULqVeB30.jpg",
        "release_date": "2019-02-14",
        "title": "Break",
        "video": false,
        "vote_average": 5.545,
        "vote_count": 157
      }
    ],
    "total_pages": 72,
    "total_results": 1440
  }