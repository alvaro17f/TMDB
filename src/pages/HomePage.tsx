import { PopularMovies } from "../components/PopularMovies";

export const HomePage = () => {
  return (
    <section className="grid md:grid-cols-4 place-items-center gap-5 m-10">
      <PopularMovies />
    </section>
  );
};
