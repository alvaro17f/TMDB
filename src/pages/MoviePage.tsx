import { useParams } from "react-router-dom";
import { MovieDetail } from "../components/MovieDetail";

export const MoviePage = () => {
  const params = useParams();

  return (
    <section className="m-5">
      <MovieDetail id={String(params.movieId)} />
    </section>
  );
};
