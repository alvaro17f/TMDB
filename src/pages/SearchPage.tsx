import { Link } from "react-router-dom";
import { useFetchSearch } from "../hooks/useFetchSearch";
import { Paginator } from "../components/Paginator";

const IMG_URL = "https://image.tmdb.org/t/p/original";

export const SearchPage = () => {
  const { data, loading, results } = useFetchSearch();

  if (loading) return <div className="text-5xl text-center">Loading ...</div>;
  if (data.length === 0)
    return <div className="text-5xl text-center">No results</div>;
  return (
    <>
      <section className="grid gap-5 m-10 md:grid-cols-4 place-items-center">
        {data?.map((movie) => (
          <Link
            id={String(movie.id)}
            to={`/movie/${movie.id}`}
            className="border rounded-xl w-56 h-[30rem]"
          >
            <img src={IMG_URL + movie.poster_path} alt={movie.title} />
            <div className="relative h-32 p-5">
              <h1 className="text-lg max-w-[200px] text-center">
                {movie.title}
              </h1>
              <p className="absolute bottom-0 text-xs text-center text-slate-700">{`Release date: ${new Date(
                movie.release_date,
              ).toLocaleDateString()}`}</p>
            </div>
          </Link>
        ))}
      </section>
      <Paginator results={results} />
    </>
  );
};