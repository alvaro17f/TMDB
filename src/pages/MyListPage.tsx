import { Link } from "react-router-dom";
import { useFetchList } from "../hooks/useFetchList";
import { useUserContext } from "../Providers/UserProvider";
import { useFetchGuest } from "../hooks/useFetchGuest";

const IMG_URL = "https://image.tmdb.org/t/p/original";

export const MyListPage = () => {
  const { data, loading } = useFetchList();
  const { request } = useFetchGuest();
  const { state } = useUserContext();

  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    event.preventDefault();
    request();
  };

  if (loading)
    return <div className="mt-5 text-5xl text-center">Loading ...</div>;
  if (!state?.guest_session_id)
    return (
      <div className="mt-5 text-2xl text-center ">
        You must click on{" "}
        <button
          type="button"
          className="p-2 border rounded-xl border-slate-400 bg-slate-300"
          onClick={handleClick}
          aria-label="get-id-list"
        >
          ðŸ˜€ Get ID{" "}
        </button>{" "}
        to see your list...
      </div>
    );
  return (
    <section className="m-5">
      <h1 className="mb-5 text-5xl">MY LIST</h1>
      <table className="w-full border ">
        <thead className="h-10 text-left bg-slate-400">
          <tr>
            <th />
            <th>Title</th>
            <th>Vote Average</th>
            <th>Personal Rating</th>
          </tr>
        </thead>
        <tbody>
          {data?.map(({ rating, title, poster_path, vote_average, id }) => (
            <tr key={id} className="border">
              <td className="w-36">
                <Link to={`/movie/${id}`}>
                  <img
                    src={IMG_URL + poster_path}
                    alt={title}
                    className="w-32"
                  />
                </Link>
              </td>
              <td>
                <Link to={`/movie/${id}`}>{title}</Link>
              </td>
              <td>{vote_average.toPrecision(2)}</td>
              <td>{rating * 2}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </section>
  );
};
