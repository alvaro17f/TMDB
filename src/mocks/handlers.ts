import { rest } from "msw";
import Popular from "./responses/Popular.json";
import Guest from "./responses/Guest.json";
import List from "./responses/List.json";
import Rating from "./responses/Rating.json";
import Search from "./responses/Search.json";
import ID from "./responses/ID.json";

export const handlers = [
  rest.get("https://api.themoviedb.org/3/movie/popular", (_req, res, ctx) => {
    return res(ctx.status(200), ctx.json(Popular));
  }),

  rest.get(
    "https://api.themoviedb.org/3/authentication/guest_session/new",
    (_req, res, ctx) => {
      return res(ctx.status(200), ctx.json(Guest));
    },
  ),

  rest.get(
    "https://api.themoviedb.org/3/guest_session/undefined/rated/movies",
    (_req, res, ctx) => {
      return res(ctx.status(200), ctx.json(List));
    },
  ),

  rest.get("https://api.themoviedb.org/3/search/movie", (_req, res, ctx) => {
    return res(ctx.status(200), ctx.json(Search));
  }),

  rest.get("https://api.themoviedb.org/3/movie/385687", (_req, res, ctx) => {
    return res(ctx.status(200), ctx.json(ID));
  }),

  rest.post(
    "https://api.themoviedb.org/3/movie/385687/rating",
    (_req, res, ctx) => {
      return res(ctx.status(200), ctx.json(Rating));
    },
  ),

  rest.get(
    "https://api.themoviedb.org/3/guest_session/ca6c96ee6faea49df91fd2f1fd48ffed/rated/movies",
    (_req, res, ctx) => {
      return res(ctx.status(200), ctx.json(List));
    },
  ),
];
