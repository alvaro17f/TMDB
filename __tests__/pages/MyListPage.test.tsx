import React from "react";
import {
  render,
  screen,
  cleanup,
  waitFor,
  fireEvent,
} from "@testing-library/react";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { afterEach, beforeEach, describe, expect, it } from "vitest";
import { MyListPage } from "../../src/pages/MyListPage";
import { MoviePage } from "../../src/pages/MoviePage";
import { server } from "../../src/mocks/server";
import { rest } from "msw";
import { UserProvider } from "../../src/Providers/UserProvider";

const wrapper = (
  <UserProvider>
    <MemoryRouter>
      <Routes>
        <Route path="/" Component={MyListPage} />
        <Route path="/movie/:movieId" Component={MoviePage} />
      </Routes>
    </MemoryRouter>
  </UserProvider>
);
describe("<MyListPage />", () => {
  beforeEach(async () => {
    render(wrapper);
    afterEach(cleanup);
  });
  it("should match the snapshot", () => {
    const { container } = render(wrapper);
    expect(container).toMatchSnapshot();
  });

  it("should render", async () => {
    screen.getByText(/loading.../i);
    await waitFor(() => {
      screen.getByText(/you must click/i);
      fireEvent.click(screen.getByLabelText(/get-id-list/i));
    });
    screen.getByText(/title/i);
    screen.getByText(/vote average/i);
    screen.getByText(/personal rating/i);
    screen.getByText(/fast x/i);
    screen.getByText(/super mario/i);
    fireEvent.click(screen.getByText(/fast x/i));
    screen.getByText(/loading/i);
    await screen.findByText(/id: 385687/i);
  });

  // it("should error", async () => {
  //   server.use(
  //     rest.get("https://api.themoviedb.org/3/movie/*", (_req, res, ctx) => {
  //       return res(ctx.status(500));
  //     }),
  //   );
  //   render(<MoviePage />);

  //   await waitFor(() => {
  //     screen.getByText(/error while fetching movie/i);
  //   });
  // });

  // it("should render Form component", async () => {
  //   screen.getByText(/you must click/i);
  //   fireEvent.click(screen.getByLabelText(/get-id-form/i));
  //   await waitFor(() => {
  //     screen.getByText(/rating/i);
  //     screen.getByRole("button", { name: /submit/i });
  //     fireEvent.click(screen.getByRole("button", { name: /submit/i }));
  //   });
  //   screen.getByText(/success/i);
  // });
});
