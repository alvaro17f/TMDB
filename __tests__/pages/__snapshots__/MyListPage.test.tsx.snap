// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MyListPage /> > should match the snapshot 1`] = `
<div>
  <div
    class="mt-5 text-5xl text-center"
  >
    Loading ...
  </div>
</div>
`;
